src/data/viteSchemaGenerator.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * As part of startup, a Json Schema file is generated based on the Zod Type for both Combos and Tricks.
+ * This is done so that the IDE can give errors on bad schemas.
+ */
+
+
+import {zodToJsonSchema} from "zod-to-json-schema"
+import { DbTricksTableZod, DbCombosTableZod } from "../lib/database/schemas/CurrentVersionSchema"
+import { writeFile } from "fs/promises"
+import { join } from "path"
+import { fileURLToPath } from "url"
+import { z } from "zod"
+
+const ShortIdArray = z.array(z.number().int().min(0))
+
+export const YamlTrickTableSchemaZod = DbTricksTableZod.omit({
+  trickStatus: true, 
+  recommendedPrerequisites: true, 
+  variationOf: true
+}).and(z.object({
+  recommendedPrerequisites: ShortIdArray.optional(),
+  variationOf: ShortIdArray.optional()
```

Comment:
It seems like you're replacing the schema definition of the `DbTricksTableZod`. I understand why you omit the trick/comboStatus but not why you're using the `ShortIdArray.xxx`.

Reply:
Basically, this is what lets you write the following in the YAML
```yaml
id: 1
...
variationOf:
  - 2
  - 3
```
instead of 
```yaml
id: 1
trickStatus: official
...
variationOf:
  - [2, official]
  - [3, official]
```
It is redundant to write "official" everywhere because we only keep official tricks in the YAML files anyways.

Reply:
Oh… I should probably read the entire comment before replying :'D

I am replacing it with ShortIdArray, which is defined as an int-Array so that we can use the modified Zod-Schema to parse the YAML files and generate the JSON schema so that your IDE can shout at you when the YAML does not meet the requirements :)